hal/hal_uv.c

```diff
@@ -50,7 +50,10 @@ adb_context_t* adb_hal_create_context(void) {
 }
 
 void adb_hal_destroy_context(adb_context_t *context) {
-    UNUSED(context);
+    adb_context_uv_t *adbd =
+        container_of(context, adb_context_uv_t, context);
+
+    uv_loop_close(adbd->loop);
```

Comment:
> That covers the case where uv_loop is initialized, but due to an error we exit without calling uv_loop_run() ?
> 
> > Releases all internal loop resources. Call this function only when the loop has finished executing and all open handles and requests have been closed, or it will return UV_EBUSY. After this function returns, the user can free the memory allocated for the loop.

As [nuttx-apps/system/adb/adb_main.c](https://github.com/apache/nuttx-apps/blob/master/system/adb/adb_main.c#L160) shown, currently `adb_hal_destroy_context()` maybe always called after `adb_hal_run()`/`uv_run()`.
```C
// L159 ~ L160
  adb_hal_run(ctx);
  adb_hal_destroy_context(ctx);
```